stl/inc/regex

```diff
@@ -3422,7 +3432,7 @@ void _Matcher3<_BidIt, _Elem, _RxTraits, _It, _Alloc>::_Increase_complexity_coun
 
 template <class _BidIt, class _Elem, class _RxTraits, class _It, class _Alloc>
 void _Matcher3<_BidIt, _Elem, _RxTraits, _It, _Alloc>::_Prepare_rep(_Node_rep* _Node) {
-    _Loop_vals_v2_t* _Psav = &_Loop_vals[_Node->_Loop_number];
+    auto* _Psav = &_Loop_vals[_Node->_Loop_number];

```

Comment:
Minor style nitpick: We generally avoid `auto*` since this doesn't really convey useful information; the pointer nature of such variables is typically obvious based on their initialization and usage. Occurs below.

This can also be top-level `const`. (`<regex>` is not very disciplined about using `const` yet so I generally don't nitpick about this, but if I'm going to push a commit, may as well add const.)